packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -82,17 +82,29 @@ impl<'id> MemoryId<'id> {
     }
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
-    pub fn clean_up<Id: Borrow<Uuid>>(id: Id) {
+    pub fn clean_up<Id: Borrow<Uuid>>(id: Id) -> Result<()> {
```

Comment:
Still not sure how I feel about this being on the `MemoryId` struct, it's strange to me that an ID has a method for removing shared memory. @TimDiekmann can you think of anywhere else this could be, would `Segment` be better? (Although it still feels like that would only be for the specific `Segment`

Reply:
I think a free-standing function in the `shared_memory` module defined in `segment.rs` would be the best way (`clean_up_by_base_id`?).  As we now have `MemoryId::base_id` I'd remove `impl Borrow<Uuid>` from `MemoryId`, too.

Reply:
Moved `clean_up` in https://github.com/hashintel/hash/pull/494/commits/99fc93e759340d83714a9bb85c5880eb7ceba3b8 and removed `Borrow<Uuid>` in https://github.com/hashintel/hash/pull/494/commits/545a9e1656de0b202aa182bf3a77ffeb93a4c30d

Reply:
There is no point having functions with `impl Borrow<Uuid>` parameters if the base struct does not implement these.

Reply:
Removed the bound on `cleanup_by_base_id` https://github.com/hashintel/hash/pull/494/commits/2a45697b9666dd122e86b76e861b76967aa824ef
Should I also remove it on `prefix`?

Reply:
Yes, please. Also, `Uuid` implements `Copy`, so you can dereference it rather than clone it (or even remove the references).

Reply:
Removed the bound and changed all references to value https://github.com/hashintel/hash/pull/494/commits/4b7cdb80051a6d4be5c4c75d03cf43808440d7d5